net/tsdial/tsdial.go

```diff
@@ -102,26 +103,55 @@ func (d *Dialer) PeerDialControlFunc() func(network, address string, c syscall.R
 	return peerDialControlFunc(d)
 }
 
-// SetDNSMap sets the current map of DNS names learned from the netmap.
-func (d *Dialer) SetDNSMap(m DNSMap) {
-	// TODO(bradfitz): update this to be aware of DNSConfig
-	// ExtraNames and CertDomains.
+// SetNetMap sets the current network map and notably, the DNS names
+// in its DNS configuration.
+func (d *Dialer) SetNetMap(nm *netmap.NetworkMap) {
```

Comment:
netmaps are not a good interchange format for configuration, because they do not blend in local preferences. I'm sad that we continue to treat it as ground truth :(

Reply:
It's getting less and less with each commit. What it really needs is the DNSConfig extra names & info on whether this node is v4, v6, or both, all of which the NetMap has. The next commit will make the Dialer start to learn about prefs (so it knows whether to use ExitDNS)